src/Sylius/Bundle/UiBundle/Renderer/TwigTemplateBlockRenderer.php

```diff
@@ -13,23 +13,29 @@
 
 namespace Sylius\Bundle\UiBundle\Renderer;
 
+use Sylius\Bundle\UiBundle\ContextProvider\ContextProviderInterface;
 use Sylius\Bundle\UiBundle\Registry\TemplateBlock;
+use Symfony\Component\DependencyInjection\ContainerInterface;
 use Twig\Environment;
+use Webmozart\Assert\Assert;
 
 /**
  * @experimental
  */
 final class TwigTemplateBlockRenderer implements TemplateBlockRendererInterface
 {
-    public function __construct(private Environment $twig)
+    public function __construct(private Environment $twig, private ContainerInterface $container)
     {
     }
 
     public function render(TemplateBlock $templateBlock, array $context = []): string
     {
+        $contextProvider = $this->container->get($templateBlock->getContextProviderClass());
+        Assert::isInstanceOf($contextProvider, ContextProviderInterface::class);
```

Comment:
I understand what is the concept here, but I believe we definitely should not go this way :dancer: I mean - way of injecting the container here. It's an obvious consequence of the design of the proposed solution, so maybe we should rearrange it a little bit. 
Let's start simple and have one `DefaultContextProvider` that can behave [as already proposed](https://github.com/Sylius/Sylius/pull/14328/files#diff-c25841738287b8332660025dab8f506a4d051c5a0d75f990eb9c477640dc6e43R20). We can maybe pass a `TemplateBlock::$name` to the `provide` function as `string $blockName` parameter, so it would be easier to provide block-specific context providing functionality. This way we avoid container injection and simplify the configuration while still keeping the possibility to extend the feature later (with, e.g. a strategy pattern for context providers). WDYT? Or am I missing something? 🖖

Reply:
Done :slightly_smiling_face:  Could you take a look now?